frontend/src/views/c-authorship.vue

```diff
@@ -449,6 +464,10 @@ export default {
       this.setInfoHash();
     },
 
+    getAuthors(file) {
+      return new Set(file.segments.map((segment) => segment.knownAuthor).filter(Boolean));
```

Comment:
I think this can be helpful. Keeping the authors sorted by the number of lines means the prominent colors in `selectedColors` will be allocated to the main authors, which at least ensures that the contributions will not be confused among the main authors. 

As for the case with more than 10 authors, I can think of 2 ways to solve it. The first is to use a tooltip that shows the author name when the mouse is hovering on the segments. The second is to add an author name column next to the line number in the colored segment, similar to what is shown by `git blame`. This can either be solved in this pull request or solved in a separate pull request.

@ckcherry23 What do you think

Reply:
Sounds great. You can go ahead with this

Reply:
@HCY123902 @zhoukerrr For sorting the authors, would it be better to sort by lines of code in that particular file, or overall lines of code in the entire repo?

Reply:
I personally suggest using the aggregate contribution in the entire repository since the color of an author should stay consistent in the same repository. What do you think @zhoukerrr

Reply:
@zhoukerrr What are my choices?

Reply:
> I personally suggest using the aggregate contribution in the entire repository since the color of an author should stay consistent in the same repository. What do you think @zhoukerrr

In the current implementation, the color assigned to an author will remain the same throughout each file in the repository, regardless of the order in which the authors are displayed.

> @zhoukerrr What are my choices?

Currently, the authors for each file are displayed in the order in which their code segments appear. 

We are considering sorting them to have the most significant contributors first, so that the more prominent colors are assigned to the most significant contributors, and we can truncate the last few if the author count is too high. We have the following choices:
1. Sort authors by lines of code in that particular file: This gives more file-level details about contributors
2. Sort authors by lines of code in the entire repository: This will keep the order in which authors appear consistent across all files.

What would you prefer? @damithc

Reply:
First, the color for a author should remain the same for every file.
As a user, my first instinct is that the author names should be sorted alphabetically, so that I can locate an author quickly in the list. Otherwise I'll have to search potentially the entire list of authors to locate an author.
It still makes sense to allocate brighter colors to high LoC authors but that can be done independent of the display order of authors, and can be a separate PR.
Also, as someone mentioned earlier, there should be a way to find the author details by hovering over the code. We should not expect the user to color-match to find the author. This is especially so because some users may be color blind.